python/pfs/pipe2d/generateReductionSpec.py

```diff
@@ -59,6 +61,88 @@ class BeamConfig:
     pfs_design_id: int
 
 
+@dataclasses.dataclass
+class SelectionCriteria:
+    """Selection criteria used in queries to opDB.
+    """
+
+    date_start: Optional[datetime.datetime] = None
+    date_end: Optional[datetime.datetime] = None
+    visit_start: Optional[int] = None
+    visit_end: Optional[int] = None
+
+    @classmethod
+    def fromNamespace(cls, args: argparse.Namespace, *, remove: bool = False) -> "SelectionCriteria":
+        """Construct SelectionCriteria from argparse.Namespace object.
+
+        Parameters
+        ----------
+        args : `argparse.Namespace`
+            A return value of ``argparse.ArgumentParser().parse_args()``.
+        remove : `bool`
+            Remove from ``args`` those members that are used by this function.
+
+        Returns
+        -------
+        criteria : `SelectionCriteria`
+            Constructed SelectionCriteria instance.
+        """
+        criteria = SelectionCriteria()
+        criteria.updateFromNamespace(args, remove=remove)
+        return criteria
+
+    def updateFromNamespace(self, args: argparse.Namespace, *, remove: bool = False):
+        """Update self using members in argparse.Namespace object.
+
+        Parameters
+        ----------
+        args : `argparse.Namespace`
+            A return value of ``argparse.ArgumentParser().parse_args()``.
+        remove : `bool`
+            Remove from ``args`` those members that are used by this function.
+        """
+        undefined = object()
+
+        for field in dataclasses.fields(self):
+            member = getattr(args, field.name, undefined)
+            if member is undefined:
+                continue
+            setattr(self, field.name, member)
+            if remove:
+                delattr(args, field.name)
+
+    def asSQL(self) -> str:
+        """Get SQL expression for queries to opDB.
+
+        Returns
+        -------
+        expression : `str`
+            SQL expression.
+        """
+        expressions = []
+        if self.date_start is not None:
+            if self.date_start.tzinfo is None:
+                datestr = self.date_start.isoformat()
+            else:
+                datestr = self.date_start.astimezone(datetime.timezone.utc).replace(tzinfo=None).isoformat()
+            expressions.append(f"pfs_visit.issued_at >= '{datestr}'")
```

Comment:
Do we need to worry about SQL injection here and below?

Reply:
For the current code, we need not worry about SQL injection because dates must have `isoformat()` method and cannot be strings. Visits must also be integers because it is converted by `:d` specifier. I will add a comment to the code in case a person carelessly modify this method.